packages/integrations/src/prowlarr/prowlarr-integration.ts

```diff
@@ -0,0 +1,83 @@
+import { Integration } from "../base/integration";
+import { IntegrationTestConnectionError } from "../base/test-connection-error";
+import type { Indexer } from "../interfaces/indexer-manager/indexer";
+import { indexerResponseSchema, statusResponseSchema } from "./prowlarr-types";
+
+export class ProwlarrIntegration extends Integration {
+  public async getIndexersAsync(): Promise<Indexer[]> {
+    const apiKey = super.getSecretValue("apiKey");
+
+    const indexerResponse = await fetch(`${this.integration.url}/api/v1/indexer?apikey=${apiKey}`);
+    if (!indexerResponse.ok) {
+      throw new Error(
+        `Failed to fetch indexers for ${this.integration.name} (${this.integration.id}): ${indexerResponse.statusText}`,
+      );
+    }
+
+    const statusResponse = await fetch(`${this.integration.url}/api/v1/indexerstatus?apikey=${apiKey}`);
+    if (!statusResponse.ok) {
+      throw new Error(
+        `Failed to fetch status for ${this.integration.name} (${this.integration.id}): ${statusResponse.statusText}`,
+      );
+    }
+
+    const indexersResult = indexerResponseSchema.array().safeParse(await indexerResponse.json());
+    const statusResult = statusResponseSchema.safeParse(await statusResponse.json());
+
+    const errorMessages: string[] = [];
+    if (!indexersResult.success) {
+      errorMessages.push(`Indexers parsing error: ${indexersResult.error.message}`);
+    }
+    if (!statusResult.success) {
+      errorMessages.push(`Status parsing error: ${statusResult.error.message}`);
+    }
+    if (!indexersResult.success || !statusResult.success) {
+      throw new Error(
+        `Failed to parse indexers for ${this.integration.name} (${this.integration.id}), most likely your api key is wrong:\n${errorMessages.join("\n")}`,
+      );
+    }
+
+    const disabledIndexerIds = new Set(statusResult.data.map((status: { id: number }) => status.id));
+
+    const indexers: Indexer[] = indexersResult.data.map((indexer) => ({
+      id: indexer.id,
+      name: indexer.name,
+      url: indexer.indexerUrls[0] ?? "",
+      enabled: indexer.enable,
+      status: disabledIndexerIds.has(indexer.id),
```

Comment:
Either the enabled flag or disabledIndexerIds name make no sense as then this would be interpreted as the same